editor-plugins.md

```diff
@@ -95,6 +95,210 @@ The diagnostics show you the syntax and semantic errors in the source code. Vari
+
+<br/>
+
+####### Starting a Program Debug Session
+
+Follow the steps below to start a program debug session.
+
+1. Select **Ballerina Debug** from the drop-down available in the upper left corner to start a program debugging session.
+
+2. Click the **Start Debugging** icon on the upper left corner to start debugging.
+
+    You view the output in the **DEBUG CONSOLE**.
+
+    ![Program Debug](/learn/images/program-debug.gif)
+
+<br/>
+
+####### Starting a Test Debug Session
+
+Follow the steps below to start a test debug session.
+
+1. Select **Ballerina Test** from the drop-down available in the upper left corner to start a test debugging session.
+
+2. Click the **Start Debugging** icon on the upper left corner to start debugging.
+
+    You view the output in the **DEBUG CONSOLE**.
+
+    ![Test Debug](/learn/images/test-debug.gif)
+
+<br/>
+
+###### Starting a Remote Debug Session
+
+Follow the steps below to start a remote debug session.
+
+1. Create the `launch.json` configuration file if it is not created already. Refer [Debug Session with Configurations](#debug-session-with-configurations) section on how to create the `launch.json` file.
+
+2. Open `launch.json` file and configure `debuggeeHost` and `debuggeePort` attributes under `Ballerina Remote` configurations section accordingly.
+
+3. After setting the remote debug configurations, select **Ballerina Remote** from the drop-down available in the upper left corner to start a remote debugging session.
+
+4. Open the Terminal and execute the Ballerina command, which you want to debug, out of the supported remote debugging commands below. 
+
+    - Debugging a Ballerina package or a single file: 
+
+    ```bash
+    bal run --debug <DEBUGGEE_PORT> <BAL_FILE_PATH/PACKAGE_PATH>
+    ```
+
+   - Debugging Ballerina executable JAR:  
+
+    ```bash 
+    bal run --debug <DEBUGGEE_PORT> <EXECUTABLE_JAR_FILE_PATH>
+    ```
+
+    - Debugging Ballerina tests: 
+
+    ```bash
+    bal test --debug <DEBUGGEE_PORT> <PACKAGE_PATH>
+    ```
+    
+    The terminal will show the following log:
+
+    ```bash
+    Listening for transport dt_socket at address: 5005
+    ```
+
+5. Click the **Start Debugging** icon on the upper left corner to start debugging.
+
+    You view the output in the **DEBUG CONSOLE**.
+
+    ![Remote Debug](/learn/images/remote-debug.gif)
+
+<br/>
+
+##### Using the Debugging Features
+
+The following debugging features are currently supported by Ballerina.
+
+- Launch/Attach
+- Breakpoints
+  - Conditional Breakpoints
+  - Logpoints
+- Pause & Continue
+- Step In/Out/Over
+- Variables
+- Call Stacks
+- Strands
+- Expression Evaluation
+
+>**Info:** For more information on debugging features using VS Code, go to the [VS Code Documentation](https://code.visualstudio.com/docs/editor/debugging).
+
```

Comment:
Shall we add this section after the `Debug Configurations` section? WDYT?

Reply:
I'll merge this. Please send fixes separately.